pkg/filesystem/subvolume.go

```diff
@@ -52,6 +57,66 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	cephclusters, err := clientsets.Rook.CephV1().CephClusters(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("failed to list CephClusters: %q", err))
+	}
+	for i := range cephclusters.Items {
+		enable = cephclusters.Items[i].Spec.External.Enable
+		if enable == true {
+			break
+		}
+	}
+	return enable
+}
+
+// getExternalClusterDetails gets the required mon-ip, id and key to connect to the
+// ceph cluster.
+func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) (string, string, string) {
+	var adminId, adminKey, m string
+
+	scList, err := clientsets.Kube.CoreV1().Secrets(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching secrets: %v\n", err))
+	}
+	for i := range scList.Items {
+		if strings.HasPrefix(scList.Items[i].ObjectMeta.Name, "rook-csi-cephfs-provisioner") {
+			data := scList.Items[i].Data
+			if data == nil {
+				logging.Fatal(fmt.Errorf("Secret data is empty"))
+			}
+			adminId = string(data["adminID"])
+			adminKey = string(data["adminKey"])
+			break
+
+		}
+	}
+
+	cm, err := clientsets.Kube.CoreV1().ConfigMaps(clusterNamespace).Get(ctx, "rook-ceph-mon-endpoints", v1.GetOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching configmaps: %v\n", err))
+	}
+
+	data := cm.Data
+	if data == nil {
+		logging.Fatal(fmt.Errorf("Configmap data is empty"))
+	}
```

Comment:
```suggestion
	data := cm.Data
	if data == nil {
		logging.Fatal(fmt.Errorf("Configmap data is empty"))
	}
```

First check len of `cm.Data` then proceed otherwise we'll get nil pointer error when it's empty